### ISSUE TITLE
JS table data not captured

### ISSUE THREAD

**Issue by umayrh** [OvidijusParsiunas/active-table#30]
Hi @OvidijusParsiunas ,

I've a table like so

<img width="714" alt="Screenshot 2024-02-02 at 5 08 18 PM" src="https://github.com/OvidijusParsiunas/active-table/assets/525221/3c739cb2-567b-4e20-996c-934b2a32e06f">

which, when exported, yields a CSV file ([table_data.csv](https://github.com/OvidijusParsiunas/active-table/files/14139882/table_data.csv)) that doesn't fully represent the data entered in the web component. I see the same problem when I access the table data using the method `elementRef.getData()`: the data is incomplete.

Any idea why?

Javascript code for table creation:

```
make_active_table = function(activeTable, tableData) {
    activeTable.data = tableData;
    activeTable.setAttribute(
        "tableStyle",
        JSON.stringify({
            "width":"100%", "boxShadow": "rgb(172 172 172 / 17%) 0px 0.5px 1px 0px", "borderRadius":"2px",
        })
    );
    activeTable.setAttribute(
        "files",
        JSON.stringify({
            "buttons": [
              {
                "import": true,
                "styles": {
                  "default": {
                    "backgroundColor": "#46acff", "height": "19px", "padding": "7px 14px", "color": "white", "border": "unset"
                  },
                  "hover": {"backgroundColor": "#2b9cf8"}, "click": {"backgroundColor": "#198eef"}
                },
                "text": "Import File",
                "position": "top-left",
                "order": 1
              },
              {
                "export": true,
                "styles": {
                  "default": {
                    "backgroundColor": "#46acff", "height": "19px", "padding": "7px 14px", "color": "white", "border": "unset"
                  },
                  "hover": {"backgroundColor": "#2b9cf8"}, "click": {"backgroundColor": "#198eef"}
                },
                "text": "Export File",
                "position": "top-left",
                "order": 0
              }
            ]
        })
    );
    return activeTable;
}

```

HTML side:

```
<div id="billTableDiv">
    <active-table id="billTable" headerStyles='{"default":{"backgroundColor": "#d6d6d630"}}'></active-table>
</div>

<button type="submit" class="btn btn-primary" onclick="beforeSubmit();">Save</button>

<script>
function beforeSubmit() {
    const invalidForm = document.querySelector('form:invalid');
    if (!invalidForm) {
        $("#spinner").modal('show');
    }
    activeTable = document.getElementById("billTable");
    document.getElementById("formField").value = JSON.stringify(activeTable.getData());
}
...
setTimeout(() => {
    billTable = make_active_table(document.getElementById("billTable"), {{ bill_table|safe }})
});
</script>
```

**Comment by umayrh** [comment]
PS. Is there anything wrong with this code?

```
                <div id="billTableDiv">
                    <active-table
                            id="billTable"
                            data="[[\"Service Date\", \"Service Code\", \"Service Category\", \"Service Name\", \"Consultant/Doctor\", \"Rate\", \"Total Quantity\", \"Total Cost\"]]"
                    ></active-table>
                </div>
```

I see an error like:

```
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'reduce')
    at InitialDataProcessing.getMaxRowLength (activeTable.bundle.js:1:277476)
    at InitialDataProcessing.preProcess (activeTable.bundle.js:1:277889)
    at ActiveTable.update (activeTable.bundle.js:870:3236)
    at ActiveTable.performUpdate (activeTable.bundle.js:16:4970)
    at ActiveTable.scheduleUpdate (activeTable.bundle.js:16:4506)
    at ActiveTable._$EP (activeTable.bundle.js:16:4414)
getMaxRowLength @ activeTable.bundle.js:1
preProcess @ activeTable.bundle.js:1
update @ activeTable.bundle.js:870
performUpdate @ activeTable.bundle.js:16
scheduleUpdate @ activeTable.bundle.js:16
_$EP @ activeTable.bundle.js:16
await in _$EP (async)
requestUpdate @ activeTable.bundle.js:16
_$Ev @ activeTable.bundle.js:16
b @ activeTable.bundle.js:16
s @ activeTable.bundle.js:37
ActiveTable @ activeTable.bundle.js:870
(anonymous) @ activeTable.bundle.js:6
__decorate @ activeTable.bundle.js:870
(anonymous) @ activeTable.bundle.js:870
```

**Comment by OvidijusParsiunas** [comment]
Hi @umayrh.

Thankyou for raising this issue. There was a bug where upon pasting data into an empty cell did not store it in memory, hence it was not exported. This has now been fixed in version `1.1.4` Please me know if you see any other issues and I will take care of them as soon as I can. Thankyou!
